### PR TITLE
chore: remove aws sdk dependency declarations

### PR DESCRIPTION
**Pull request by sktimalsina**
- [x] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*
remove dependencies in legacy sdk

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/build.gradle [hunk 1]
```diff
@@ -42,13 +41,17 @@ dependencies {
...
     testImplementation dependency.junit
     testImplementation dependency.mockito
     testImplementation dependency.mockk
     testImplementation dependency.robolectric
     testImplementation dependency.androidx.test.core
     testImplementation dependency.kotlin.reflection
 
+    androidTestImplementation dependency.gson
+    //noinspection GradleDependency
+    androidTestImplementation "com.amazonaws:aws-android-sdk-core:2.55.0"
```
Comment (div5yesh) [inline aws-auth-cognito/build.gradle:54]: 🙌

**Review by div5yesh — APPROVED** [review]

File: aws-auth-cognito/build.gradle (hunk 1 above)
Comment (tjleing) [reply aws-auth-cognito/build.gradle:54]: 🙌

File: aws-auth-cognito/build.gradle [hunk 2]
```diff
@@ -30,7 +30,6 @@ dependencies {
     implementation dependency.androidx.security
     implementation dependency.androidx.browser
 
-    implementation dependency.aws.core
     implementation dependency.aws.cognitoidentity
     implementation dependency.aws.cognitoidentityprovider
```
Comment (tjleing) [inline aws-auth-cognito/build.gradle:34]: are these still Android SDK dependencies?

**Review by tjleing — APPROVED** [review]

File: aws-auth-cognito/build.gradle (hunk 2 above)
Comment (sktimalsina) [reply aws-auth-cognito/build.gradle:34]: No, we should have done better job of renaming them as I deleted them at first 🤦

**Pull request merged**